src/rust/ensogl/lib/list-view/src/entry.rs

```diff
@@ -114,6 +120,99 @@ impl ModelProvider for EmptyProvider {
 }
 
 
+// === Model Provider for Vectors ===
+
+/// An Entry Model Provider that wraps a Vec<T:Into<Model>>.
+#[derive(Clone,Debug)]
+pub struct VectorProvider<T> {
+    content : Vec<T>,
+}
```

Comment:
Why do we need wrapper? Cannot we just `impl<T:Into<Model> + Debug + Clone> ModelProvider for Vec<T>`?

Reply:
Yes! Did that and removed the wrapper.